File: AlgorithmReview/Programmers/Lv2/영어 끝말잇기.swift
//
//  영어 끝말잇기.swift
//  AlgorithmReview
//
//  Created by 박소윤 on 2023/03/24.
//

import Foundation

func solution(_ n:Int, _ words:[String]) -> [Int] {
    var checkWords = Set<String>()
    var ans = [0,0]
    for (i,c) in words.enumerated(){
        if(checkWords.contains(c) || (i >= 1 && c.first! != words[i-1].last!)){
            ans = [i % n + 1, i / n + 1]; break
        }
        checkWords.insert(c)
    }
    return ans
}

File: Python/영어 끝말잇기.py
def solution(n, words):

    speak = set()
    for (i, w) in enumerate(words):
        if w in speak or (i > 0 and words[i-1][-1] != w[0]):
            return [int(i%n)+1, int(i/n)+1]
        speak.add(w)
    
    return [0,0]
